catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitor.java

```diff
@@ -151,143 +162,134 @@ public void setAttributeOverrides(List<String> attributeOverrides) {
         return this.attributeOverrides;
     }
 
+    public List<RouteDefinition> getRouteDefinitions() {
+        return camelContext.getRouteDefinitions();
+    }
+
+    /*
+        Starts a task that waits for the "content" CamelComponent before adding the routes from
+        the RouteBuilder to the CamelContext. This ensures CDMs will automatically start after
+        a system shutdown.
+     */
     private void configureCamelRoute() {
         LOGGER.trace("ENTERING: configureCamelRoute");
-
-        // Must have a directory to be monitored to be able to configure the Camel route.
         if (StringUtils.isEmpty(monitoredDirectory)) {
-            LOGGER.debug("Cannot setup camel route - must specify a directory to be monitored");
+            LOGGER.warn("Cannot setup camel route - must specify a directory to be monitored");
             return;
         }
+        CompletableFuture.runAsync(this::attemptAddRoutes);
+        LOGGER.trace("EXITING: configureCamelRoute");
+    }
 
-        RouteBuilder routeBuilder = new RouteBuilder() {
-            @Override
-            public void configure() throws Exception {
-                String inbox = "file:" + monitoredDirectory + "?moveFailed=.errors";
-                if (copyIngestedFiles) {
-                    inbox += "&move=.ingested";
-                } else {
-                    inbox += "&delete=true";
-                }
-                LOGGER.debug("inbox = {}", inbox);
-
-                RouteDefinition routeDefinition = from(inbox);
-
-                if (attributeOverrides != null) {
-                    String attributeOverrideString = attributeOverrides.stream()
-                            .map(String::trim)
-                            .collect(Collectors.joining(","));
-                    routeDefinition.setHeader(Constants.ATTRIBUTE_OVERRIDES_KEY,
-                            simple(attributeOverrideString));
-                }
-
-                routeDefinition.process(systemSubjectBinder)
-                        .to("content:framework");
-            }
-        };
-
+    /*
+        The task method used to verify the "content" CamelComponent before loading routes.
+     */
+    private void attemptAddRoutes() {
+        LOGGER.debug("Attempting to add routes for CDM {}", runtimeMonitorId);
```

Comment:
Do not use acronyms in log messages unless they are very well known.

I agree that the directory path would be more informative.